.github/workflows/macos_build.yml

```diff
@@ -0,0 +1,23 @@
+name: Macos Build
+
+on: [push, pull_request, workflow_dispatch]
+concurrency:
+  group: ci-${{github.workflow}}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: 'macos-latest'
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Prerequisites
+        shell: bash
+        run: |
+          Tools/environment_install/install-prereqs-mac.sh -y
+          source ~/.bash_profile
+          echo $PATH
+          ./waf configure --board CubeOrange
```

Comment:
We should make a matrix and build all vehicles, that way we profit of a new check

Reply:
@khancyr feel free to push any required changes to this workflow. I have already given you push access to my repo.

Reply:
There is also ccache available on macos, we can probably use that as well.